payu/models/model.py

```diff
@@ -286,6 +286,12 @@ def setup(self):
             # If have exe manifest this implies exe reproduce is True. Do not
             # want to overwrite exe manifest in this case
             if not self.expt.manifest.have_manifest['exe']:
+                # Check whether executable path exists
```

Comment:
I'm slightly concerned about putting the checks here, even though I suggested this location I think I was wrong. 

The tests are only done if there is not an existing manifest. An existing manifest means we want to check for reproducibility. It will do the hash checks, which would fail if it was missing, but would not detect a change in file permissions.

I'd suggest moving your tests above line 288:
```python
if not self.expt.manifest.have_manifest['exe']:
```

Reply:
I initially had the check above line 288. But it would fail this [manifest test](https://github.com/payu-org/payu/blob/edf8d8c538040447dcd3bdd944cba5bab86937c1/test/test_manifest.py#L217C1-L230) which makes the executable unfindable but tests that it still works because it has the same name as executable in the manifest. Do you think it would it be ok if I changed/removed that test?

Reply:
Ok, maybe I've changed my mind *again* and we should remove that test, as it is testing to make sure the manifest is still correct even if the `config.yaml` has been changed so the paths to the executables is broken. In retrospect I don't think we should support that use case, and I doubt anyone is relying on it.

There is a tension between where the source of truth lies. It was always in `config.yaml`, but the manifests now also hold information, and when we want to reproduce a run, or make sure the executables haven't changed then the manifests become the source of truth. I think I need to think about this a bit more!

Reply:
Ok, I agree with not supporting invalid paths in `config.yaml` as that will at least help catch any mistakes. Unless it was practice to put invalid paths in to check that exe paths in the manifests were used? But in that case should the practice be to remove the exe path from the config.yaml?